src/main/java/dtb/lego/truck/rest/component/events/entity/events/Event.java

```diff
@@ -0,0 +1,30 @@
+package dtb.lego.truck.rest.component.events.entity.events;
+
+import dtb.lego.truck.rest.component.events.entity.Components;
+import org.springframework.data.repository.NoRepositoryBean;
+
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+
+/**
+ * Event interface used to outline all the functions to be
+ * implemented by the different kinds of events
+ */
+@NoRepositoryBean
+@MappedSuperclass
+public class Event extends Object {
+    @Id
+    Long timestamp;
+    Components name;
+
+    public Event(long timestamp) {
+        this.timestamp = timestamp;
+    }
+
+    public Event() {
```

Comment:
- You don't need to declare an empty constructor here. Empty c'tors are created automatically for every class w/ no ctors.
- Every Java class inherits from Object. There's no need to explicitly declare it.

Reply:
It doesn't create automatically since I already have another constructor on the class, but the hibernate needs a no-argument constructor since it uses reflection to build the Entity.